service/service.py

```diff
@@ -88,6 +88,44 @@ def list_promotions():
     app.logger.info("Returning %d promotions", len(results))
     return make_response(jsonify(results), status.HTTP_200_OK)
 
+######################################################################
+# UPDATE AN EXISTING PROMOTION
+######################################################################
+@app.route("/promotions/<int:promotion_id>", methods=["PUT"])
+def update_promotions(promotion_id):
+    """
+    Update a Promotion
+    This endpoint will update a Promotion based the body that is posted
+    """
+    app.logger.info("Request to update promotion with id: %s", promotion_id)
+    check_content_type("application/json")
+    promotion = Promotion.find(promotion_id)
+    if not promotion:
+        raise NotFound("Promotion with id '{}' was not found.".format(promotion_id))
+    promotion.deserialize(request.get_json())
+    promotion.id = promotion_id
+    promotion.update()
+    app.logger.info("Promotion with ID [%s] updated.", promotion.id)
+    return make_response(jsonify(promotion.serialize()), status.HTTP_200_OK)
+
+######################################################################
+# DELETE A PROMOTION
+######################################################################
+@app.route("/promotions/<int:promotion_id>", methods=["DELETE"])
+def delete_promotions(promotion_id):
+    """
+    Delete a Promotions
+    This endpoint will delete a Promotion based the id specified in the path
+    """
+    app.logger.info("Request to delete promotion with id: %s", promotion_id)
+    promotion = Promotion.find(promotion_id)
+    if not promotion:
+        return make_response("", status.HTTP_200_OK)
```

Comment:
This should be a 404 status

Reply:
Hi, Vicky, thank you for your commit. The acceptance criteria of the issue sets that if you call the DELETE function on a promotion that doesn't exist, should return OK, so I set this to 200_OK status. If it is better to set to 404 status, do we need to set a fix to the issue then change it, or could we just change it in the same issue?

Reply:
Ah I missed that, you're right! Nevermind :sweat_smile:

Reply:
😊